src/redux/publisherConsentsSlice.js

```diff
@@ -0,0 +1,62 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import { getConsents } from 'api';
+
+const initialState = {
+  publisherConsents: [],
+  checked: [],
+  loading: false,
+  error: ''
+};
+
+export const fetchPublisherConsents = createAsyncThunk(
+  'publisherConsents',
+  async (publisherId, { rejectWithValue }) => {
+    try {
+      const responseData = await getConsents(publisherId);
+      return responseData;
+    } catch (error) {
+      return rejectWithValue(error.message);
+    }
+  }
+);
+
+export const consentsSlice = createSlice({
+  name: 'publisherConsents',
+  initialState,
+  reducers: {
+    setChecked(state, { payload }) {
+      state.checked[payload] = !state.checked[payload];
+    }
+  },
+  extraReducers: {
+    [fetchPublisherConsents.pending]: state => {
+      state.loading = true;
+    },
+    [fetchPublisherConsents.fulfilled]: (state, { payload }) => {
+      if (payload.responseData && payload.responseData.consents) {
+        const publisherConsents = payload.responseData.consents.map(element => {
+          return {
+            name: element.name,
+            label: element.label,
+            version: element.version,
+            required: element.required
+          };
+        });
```

Comment:
the whole 'publisherConsents ' could be simplified to state.publisherConsents = payload